src/Cli/dotnet/Program.cs

```diff
@@ -134,93 +119,68 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient = null
-                    {
-                        Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, bool.TrueString);
-                        CommandContext.SetVerbose(true);
-                    }
-                    else if (IsArg(args[lastArg], "version"))
-                    {
-                        PrintVersion();
-                        return 0;
-                    }
-                    else if (IsArg(args[lastArg], "info"))
-                    {
-                        PrintInfo();
-                        return 0;
-                    }
-                    else if (IsArg(args[lastArg], "h", "help") ||
-                             args[lastArg] == "-?" ||
-                             args[lastArg] == "/?")
-                    {
-                        HelpCommand.PrintHelp();
-                        return 0;
-                    }
-                    else if (args[lastArg].StartsWith("-", StringComparison.OrdinalIgnoreCase))
+                    Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, bool.TrueString);
+                    CommandContext.SetVerbose(true);
+                    Reporter.Reset();
+                }
+                if (parseResult.HasOption(Parser.VersionOption))
+                {
+                    CommandLineInfo.PrintVersion();
+                    return 0;
+                }
+                else if (parseResult.HasOption(Parser.InfoOption))
+                {
+                    CommandLineInfo.PrintInfo();
+                    return 0;
+                }
+                else if (parseResult.CommandResult.Command.Equals(Parser.RootCommand) && parseResult.HasOption("-h"))
+                {
+                    HelpCommand.PrintHelp();
+                    return 0;
+                }
+                else
+                {
+                    PerformanceLogEventSource.Log.FirstTimeConfigurationStart();
+
+                    var environmentProvider = new EnvironmentProvider();
+
+                    bool generateAspNetCertificate =
+                        environmentProvider.GetEnvironmentVariableAsBool("DOTNET_GENERATE_ASPNET_CERTIFICATE", defaultValue: true);
+                    bool telemetryOptout =
+                      environmentProvider.GetEnvironmentVariableAsBool("DOTNET_CLI_TELEMETRY_OPTOUT", defaultValue: false);
+                    bool addGlobalToolsToPath =
+                        environmentProvider.GetEnvironmentVariableAsBool("DOTNET_ADD_GLOBAL_TOOLS_TO_PATH", defaultValue: true);
+                    bool nologo =
+                        environmentProvider.GetEnvironmentVariableAsBool("DOTNET_NOLOGO", defaultValue: false);
+
+                    ReportDotnetHomeUsage(environmentProvider);
+
+                    var isDotnetBeingInvokedFromNativeInstaller = false;
+                    if (parseResult.CommandResult.Command.Equals(Parser.InstallSuccessCommand))
```

Comment:
equals works here? Does system commandline have special handing for command being equal?

Reply:
so i assume system.commandline has special handing for command being equal

Reply:
Not that I'm aware of, I changes this to check that the names are equal in one of my later commits.